include/jwt-cpp/jwt.h

```diff
@@ -2668,6 +2668,162 @@ namespace jwt {
+	template<typename json_traits>
+	class jwks {
+		using basic_claim_t = basic_claim<json_traits>;
+		template<typename> friend class jwks_keys;
+
+	public:
+		JWT_CLAIM_EXPLICIT jwks(const typename json_traits::string_type& jwks_token) {
+			auto parse_claims = [](const typename json_traits::string_type& str) {
+				using basic_claim_t = basic_claim<json_traits>;
+				std::unordered_map<typename json_traits::string_type, basic_claim_t> res;
+				typename json_traits::value_type val;
+				if (!json_traits::parse(val, str))
+					throw std::runtime_error("Invalid json");
+
+				for (const auto& e : json_traits::as_object(val)) {
+					res.emplace(e.first, basic_claim_t(e.second));
+				}
+
+				return res;
+			};
+
+			jwks_claims = parse_claims(jwks_token);
+		}
+
+		/**
+		 * Get algorithm claim
+		 * \return algorithm as string
+		 * \throw std::runtime_error If claim was not present
+		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
+		 */
+		typename json_traits::string_type get_algorithm() const { return get_jwks_claim("alg").as_string(); }
+
+		/**
+		 * Get key id claim
+		 * \return key id as string
+		 * \throw std::runtime_error If claim was not present
+		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
+		 */
+		typename json_traits::string_type get_key_id() const { return get_jwks_claim("kid").as_string(); }
+
+		/**
+		 * Get x5c claim
+		 * \return x5c as string
+		 * \throw std::runtime_error If claim was not present
+		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
+		 */
+		typename json_traits::string_type get_x5c() const { return json_traits::as_string(get_jwks_claim("x5c").as_array().front()); } //do not use serialize() instead
+
+		/**
+		 * Check if algortihm is present ("alg")
+		 * \return true if present, false otherwise
+		 */
+		bool has_algorithm() const noexcept { return has_jwks_claim("alg"); }
+
+		/**
+		 * Check if key id is present ("kid")
+		 * \return true if present, false otherwise
+		 */
+		bool has_key_id() const noexcept { return has_jwks_claim("kid"); }
+
+		/**
+		 * Check if x5c is present ("x5c")
+		 * \return true if present, false otherwise
+		 */
+		bool has_x5c_id() const noexcept { return has_jwks_claim("x5c"); }
+
+		/**
+		 * Check if a jwks claim is present
+		 * \return true if claim was present, false otherwise
+		 */
+		bool has_jwks_claim(const typename json_traits::string_type& name) const noexcept { return jwks_claims.count(name) != 0; }
+
+		/**
+		 * Get jwks claim
+		 * \return Requested claim
+		 * \throw std::runtime_error If claim was not present
+		 */
+		basic_claim_t get_jwks_claim(const typename json_traits::string_type& name) const {
+			if (!has_jwks_claim(name)) {
+				typename json_traits::string_type error_msg = "claim \"";
+				error_msg += name;
+				error_msg += "\" not found";
+				throw std::runtime_error(error_msg.c_str());
```

Comment:
Since all of of them should be changed in the future...

```suggestion
				throw std::runtime_error("claim not found");
```